pkg/hubble/relay/observer/server.go

```diff
@@ -231,6 +232,52 @@ func (s *Server) GetNodes(ctx context.Context, req *observerpb.GetNodesRequest)
 	return &observerpb.GetNodesResponse{Nodes: nodes}, nil
 }
 
+// GetNamespaces implements observerpb.ObserverClient.GetNamespaces.
+func (s *Server) GetNamespaces(ctx context.Context, req *observerpb.GetNamespacesRequest) (*observerpb.GetNamespacesResponse, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		ctx = metadata.NewOutgoingContext(ctx, md)
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
Is this needed? It looks like the new context is only passed to the goroutines started with `g.Go` and we do wait for all of them to return with `g.Wait()` before the function returns.
```suggestion
```

Reply:
I was just following the already established pattern. I believe the intent here is to ensure the context is cancelled when we return from the function. However, since we're waiting on the errgroup to return for all running routines, your right, we do not need this.